adding BCD for events mixed bag part 2

chrisdavidmills opened the pull request:
As per https://github.com/mdn/sprints/issues/1126#issuecomment-487053382

The following is a list describing where the data comes from, for each addition.

* api.XMLHttpRequest.timeout_event: Copied from api.XMLHttpRequestEventTarget.ontimeout.
* api.HTMLSlotElement.slotchange_event: Copied from api.HTMLSlotElement.assignedNodes, as I remember this stuff being supported at roughly the same time, although it would be very hard to find out definitively. There is no onslotchange to copy from; the property doesn't exist.
* api.Element.select_event: Copied from api.GlobalEventHandlers.onselect.
* api.Element.show_event: Copied from api.GlobalEventHandlers.onshow.
* api.Performance.resourcetimingbufferfull_event: Copied from api.Performance.onresourcetimingbufferfull.
* api.Document.scroll_event and api.Element.scroll_event: Copied from api.GlobalEventHandlers.onscroll.
* api.Document.wheel_event and api.Element.wheel_event: Copied from api.GlobalEventHandlers.onwheel.
* api.Window.resize_event: Copied from api.GlobalEventHandlers.onresize.
* api.Window.storage_event: Copied from api.WindowEventHandlers.onstorage.
* api.Window.popstate_event: Copied from api.WindowEventHandlers.onpopstate.

jpmedley commented on api/Window.json:
```diff
@@ -6816,6 +6816,55 @@
+              "version_added": "5.1"
+            },
+            "webview_android": {
+              "version_added": null
```
This should be `true`.

jpmedley commented on api/Window.json:
```diff
@@ -9504,6 +9605,55 @@
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Window/storage_event",
+          "support": {
+            "chrome": {
+              "version_added": null
```
This should be `false`.

wbamberg commented on api/Element.json:
```diff
@@ -6913,6 +7017,58 @@
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
```
This is deprecated according to the page: https://developer.mozilla.org/en-US/docs/Web/API/Element/show_event

chrisdavidmills commented:
@jpmedley @wbamberg Thanks for your reviews gents.

I've:

* updated the data values according to Joe's comments.
* set show_event deprecated to true
* Removed the note stuff from the popstate page and added the equivalent information to BCD as notes.

wbamberg commented:
@jpmedley , please can you check that your comments are addressed? I'd love to be able to merge this PR so we can close https://github.com/mdn/sprints/issues/1126.

chrisdavidmills commented:
Sorry for dragging this one out, but I decided that I wasn't sure whether the popstate behavior was incorrect, so I removed those words and added a firefox note.

I also read https://bugzilla.mozilla.org/show_bug.cgi?id=1528052 and decided to add some notes around that

wbamberg commented on api/Window.json:
```diff
@@ -7564,19 +7568,23 @@
             "opera": {
-              "version_added": "32"
+              "version_added": "32",
+              "notes": "Opera does not emit a <code>resize</code> event on page load."
```
I don't know what the convention here is, but I try to avoid naming browsers in notes, since that is clear from the context, and this stops note text from being the same, and that can be helpful to code that renders tables. So I would say just "The \<code>\<resize>\</code> event is not emitted on page load".

Would happily hear from @Elchi3  on this though.

vinyldarkscratch replied:
We've included the browser names in notes on several other features, especially when including a version number.  I don't really think we have a convention one way or the other.

wbamberg replied:
Yes, fair enough!

vinyldarkscratch replied:
Personally, I'm in favor of keeping the browser name in the notes.  It helps maintain context if the note is quoted anywhere else.  Perhaps we should open a discussion and define a specific convention?

chrisdavidmills commented:
> I don't in general like documentation that uses words like "it will soon stop doing so", because "soon" is a moving target. If bug 1528052 sticks, this will get out of date almost immediately, and if it doesn't, this is just wrong. I think we should omit that part and deal with it in the dev-doc needed update for that bug.

Yeah, I think you've got a good point here — I've removed those bits.

I did however keep the browser names in the notes; I prefer that style.

wbamberg commented:
Thanks for the updates, @chrisdavidmills ! I'm happy with this.
